packages/client/hooks/useAuthRoute.ts

```diff
@@ -52,13 +52,14 @@ const useAuthRoute = (inOptions: Options = {}) => {
   useEffect(checkAuth, [atmosphere.authObj, history, options])
 
   // Detect changes to the auth token in localStorage (e.g., user signs out from another tab)
-  const token = window.localStorage.getItem('Action:token')
   useEffect(() => {
-    if (!token) {
-      atmosphere.setAuthToken(null)
+    window.addEventListener('storage', (e) => {
```

Comment:
going to e.g. /page/123 would redirect to /redirectTo=/page/123 and then redirect _again_ to `/`, getting rid of the search params. this is because checkAuth was getting called twice.